### ISSUE TITLE
[P3] Barge-in support (interrupt TTS playback)

### ISSUE THREAD

**Issue by drizki** [drizki/voxclaw#110]
## Goal
Allow users to interrupt TTS playback by speaking (barge-in).

## Current Behavior
When TTS is playing, user speech is ignored until playback completes.

## Proposed Behavior
If VAD detects speech during TTS playback:
1. Stop TTS playback immediately
2. Capture and transcribe the interruption
3. Process as new input

### 1. State Machine Update
Current states: `Idle → Listening → Processing → Speaking → Idle`

Add transition:
```
Speaking + SpeechStart → Listening (cancel playback)
```

### 2. AudioPlayback Changes
```rust
impl AudioPlayback {
    pub fn stop(&self);  // Immediate stop
    pub fn is_playing(&self) -> bool;
}
```

### 3. VAD During Playback
- Keep VAD running during Speaking state
- Filter out TTS audio from mic (echo cancellation helps)
- Detect external speech vs playback bleed

### 4. Configuration
```toml
[chat]
barge_in = true
barge_in_threshold = 0.7  # VAD confidence to trigger
```

### 5. Protocol Update
```json
{"type": "playback_interrupted", "reason": "user_speech"}
```

## Challenges
- Echo cancellation (mic picks up speaker output)
- False positives from background noise
- Partial response handling

## Acceptance Criteria
- [ ] Speaking while TTS playing stops playback
- [ ] Interrupted speech is transcribed
- [ ] Configurable on/off
- [ ] Works with local transport
- [ ] Reasonable false positive rate

## References
- Current state machine: `src/state/mod.rs`
- Similar: Amazon Alexa barge-in, Google Assistant
